test/normalize.test.ts

```diff
@@ -25,7 +25,7 @@ describe('normalizeURL', () => {
     '/http:/': '/http:/',
     'http://[2001:db8:85a3:8d3:1319:8a2e:370:7348]/': 'http://[2001:db8:85a3:8d3:1319:8a2e:370:7348]/',
     'http://localhost/?redirect=http://google.com?q=test': 'http://localhost/?redirect=http://google.com?q=test',
-    'http://localhost/?email=some+v1@email.com': 'http://localhost/?email=some%2Bv1@email.com',
+    'http://localhost/?email=some+v1@email.com': 'http://localhost/?email=some+v1@email.com',
```

Comment:
Note.

Reply:
We need to preserve it i think otherwise we bring back another regression that when values are decoded, they will contain space 😆 (god bless vue-route that has plus encoding)

Reply:
Will return to this shortly. Could you give a specific instance of where the regression would be?
```
> ufo.parsePath('http://localhost/?email=some+v1@email.com')
{
  pathname: 'http://localhost/',
  search: '?email=some+v1@email.com',
  hash: ''
}
> ufo.parseURL('http://localhost/?email=some+v1@email.com')
{
  protocol: 'http:',
  auth: '',
  host: 'localhost',
  pathname: '/',
  search: '?email=some+v1@email.com',
  hash: ''
}
> new ufo.$URL('http://localhost/?email=some+v1@email.com')
$URL {
  query: { email: 'some v1@email.com' },
  protocol: 'http:',
  host: 'localhost',
  auth: '',
  pathname: '/',
  hash: ''
}
```

Reply:
I think I see what you mean. The issue is that if `normalizeURL` will encode `+` =>  `%2B` then it can't be called on previously normalized URLs; i.e. it is a one-time action.

The behaviour in this PR *should* match Web APIs. For example:

```js
const p = new URLSearchParams('email=some+v1@email.com')
p.get('email')
// "some v1@email.com"
```